app/components/UI/Stake/components/StakingEarnings/index.tsx

```diff
@@ -3,49 +3,60 @@ import { View } from 'react-native';
 import Text, {
   TextColor,
   TextVariant,
-} from '../../../../component-library/components/Texts/Text';
-import { useStyles } from '../../../../component-library/hooks';
+} from '../../../../../component-library/components/Texts/Text';
+import { useStyles } from '../../../../../component-library/hooks';
 import styleSheet from './StakingEarnings.styles';
 import {
   IconColor,
   IconName,
-} from '../../../../component-library/components/Icons/Icon';
+} from '../../../../../component-library/components/Icons/Icon';
 import ButtonIcon, {
   ButtonIconSizes,
-} from '../../../../component-library/components/Buttons/ButtonIcon';
-import useTooltipModal from '../../../../components/hooks/useTooltipModal';
-import { strings } from '../../../../../locales/i18n';
-import { isPooledStakingFeatureEnabled } from '../../Stake/constants';
-
-// TODO: Remove mock data when connecting component to backend.
-const MOCK_DATA = {
-  ANNUAL_EARNING_RATE: '2.6%',
-  LIFETIME_REWARDS: {
-    FIAT: '$2',
-    ETH: '0.02151 ETH',
-  },
-  EST_ANNUAL_EARNINGS: {
-    FIAT: '$15.93',
-    ETH: '0.0131 ETH',
-  },
-};
-
-const StakingEarnings = () => {
-  // TODO: Remove mock data when connecting component to backend.
-  const { ANNUAL_EARNING_RATE, LIFETIME_REWARDS, EST_ANNUAL_EARNINGS } =
-    MOCK_DATA;
+} from '../../../../../component-library/components/Buttons/ButtonIcon';
+import useTooltipModal from '../../../../../components/hooks/useTooltipModal';
+import { strings } from '../../../../../../locales/i18n';
+import { isPooledStakingFeatureEnabled } from '../../../Stake/constants';
+import useStakingEligibility from '../../hooks/useStakingEligibility';
+import useStakingChain from '../../hooks/useStakingChain';
+import { StakeSDKProvider } from '../../sdk/stakeSdkProvider';
+import useStakingEarnings from '../../hooks/useStakingEarnings';
+import usePooledStakes from '../../hooks/usePooledStakes';
 
+const StakingEarningsContent = () => {
   const { styles } = useStyles(styleSheet, {});
 
   const { openTooltipModal } = useTooltipModal();
 
+  const { hasStakedPositions } = usePooledStakes();
+
+  const {
+    annualRewardRate,
+    lifetimeRewardsETH,
+    lifetimeRewardsFiat,
+    estimatedAnnualEarningsETH,
+    estimatedAnnualEarningsFiat,
+    isLoadingEarningsData,
+  } = useStakingEarnings();
+
   const onNavigateToTooltipModal = () =>
     openTooltipModal(
       strings('stake.annual_rate'),
       strings('tooltip_modal.reward_rate.tooltip'),
     );
 
-  if (!isPooledStakingFeatureEnabled()) return <></>;
+  const { isEligible, isLoadingEligibility } = useStakingEligibility();
+
+  const { isStakingSupportedChain } = useStakingChain();
+
+  const isLoadingData = isLoadingEligibility || isLoadingEarningsData;
```

Comment:
By the name of the variable I believe you mean && instead of || right? The data is only loading if it's eligble? I can be missing some context, but I think it's safe to assume that if is not eligible is never loading if it's eligible can be loading? 
Let me know if I'm missing some context here
```suggestion
  const isLoadingData = isLoadingEligibility && isLoadingEarningsData;
```

Reply:
Good catch !

Reply:
The isLoadingData is checking whether any of these eligibility functions and data functions results are still being fetched . 
If any of them are still loading then we don't show the data to the user .

Reply:
Thanks for the context!